src/helpers/brewings.ts

```diff
@@ -0,0 +1,116 @@
+import { InstructionLogs, StatusLogs } from '@prisma/client';
+import { timeinterval } from '../brewing';
+import { BrewingApi, BrewingRaw, BrewinsgRaw } from '../types/Brewing';
+import queryErrorHanlder from '../queryErrorHandler';
+import db from '../prismaClient';
+import { formatRecipe } from './recipe';
+
+export const setBrewingState = async (id: number, state: string) => {
+  try {
+    await db.brewings.update({
+      where: {
+        id,
+      },
+      data: {
+        state,
+      },
+    });
+  } catch (e) {
+    queryErrorHanlder(e, 'Brewing state update query');
+  }
+};
+export const getAllBrewingsQuery = async () => {
+  return db.brewings.findMany({
+    where: {
+      NOT: {
+        state: 'Active',
+      },
+    },
+  });
+};
+
+export const getBrewingQuery = async (id: number) => {
+  return db.brewings.findUnique({
+    where: { id },
+    include: {
+      Recipes: {
+        include: {
+          Ingredients: true,
+          Instructions: {
+            orderBy: {
+              ordering: 'asc',
+            },
+            include: {
+              FunctionTemplates: true,
+              FunctionOptions: {
+                select: {
+                  codeName: true,
+                },
+              },
+              Blocks: {
+                select: {
+                  name: true,
+                  id: true,
+                },
+              },
+            },
+          },
+        },
+      },
+      InstructionLogs: true,
+      StatusLogs: true,
+    },
+  });
+};
+
+export const processBrewing = (brewing: BrewingRaw) => {
+  const brew: BrewingApi = {
+    id: brewing.id,
+    notes: brewing.notes,
+    evaluation: brewing.evaluation,
+    endState: brewing.state,
+    startedAt: brewing.createdAt,
+    finishedAt: brewing.updatedAt ? brewing.updatedAt : null,
+    recipe: formatRecipe(brewing.Recipes),
+    InstructionLogs: brewing.InstructionLogs.map((elem: InstructionLogs) => {
+      return {
+        id: elem.id,
+        brewingId: elem.brewingId,
+        instructionId: elem.instructionId,
+        finished: elem.finished,
+        startedAt: elem.createdAt.getTime() - brewing.createdAt.getTime(),
+      };
+    }),
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    StatusLogs: brewing.StatusLogs.filter((val, indx) => {
+      const logsPerMinute = Math.floor(60000 / timeinterval);
```

Comment:
mozno by som tam dal trosku menej tu periodu, staci dat raz za 15 sekund

lepsie sa nam to bude aj testovat ze nemusi ti bezat recept tak dlho aby si to mohol potom vidiet

Reply:
No periodu na logovanie stavu mame na 1s ale s tym sa da hrat v ramci brewingu to je ten importnuty timeinterval. Tu fitrujem aby pre zobrazenie historie a tych grafov islo iba jedno varenie lebo mat 60 merani za minutu je zbytocne pre zobrazenie v grafe a aj by to asi dlho srotilo tie grafy s tolkymi meraniami a ak bude recept trvat hodiny tak posielat tisicovi json na parseovanie bude tiez trvat par minut.